docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+3. Missed messages are stored into local database
+4. Community owner node creates message archive and magnet link for missed messages
+5. Community oowner node creates new message archive index bundled with newly created magnet link
+6. The new index is is distributed to community members via special channel through the Waku network
+
+### Receiving community history archives
+
+Community member nodes go through the following (high level) process to fetch and restore community message histories:
+
+1. User joins community and becomes community member
+2. By joining a community, member nodes automatically subscribe to special magnet link channel provided by the community
+3. Member node requests message history (last 30 days) of community channels from store nodes
+4. Member node receives magnet link message from store nodes
+5. Member node extracts magnet link from message and passes it to torrent client
+6. Torrent client downloads latest message archive index via magnet link
+7. Member node fetches missing archives via torrent
+8. Member node unpacks and decompresses message archive data to then hydrate its local database
+
+## Storing live messages
+
+Community owner nodes MUST store live messages as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/). This is required to provide confidentiality, authenticity, and integrity of message data distributed via the BitTorrent layer, and later validated by Status nodes when they unpack message history archives.
+
+Community owner nodes SHOULD remove those messages from their local databases after they have been turned into archives and distributed to the BitTorrent network.
+
+## Exporting messages for bundling
+
+Community owner nodes export messages from their local database for creating and bundling history archives using the following criteria:
+
+- Messages to be exported MUST have a `contentTopic` that matches any of the topics of the community channels
+- Messages to be exported MUST have a `timestamp` that lies within a given time range
+
+The range for the `timestamp` depends on the context in which the community owner node attempts to create a history archive. This can be one of the following:
+
+1. The community owner node attempts to create an archive periodically for the past seven days (including the current day). In this case, the `timestamp` has to lie within the day the last time an archive was created and the current day.
+2. The community owner node has been offline and attempts to create an archive for all the live messages it has missed since it went offline. In this case, the `timestamp` has to lie within the day the latest message was received and the current day.
+
+Exported messages MUST be restored as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/) for bundling. Waku messages that have been exported for bundling can now be removed from the community owner node's database (community owner nodes still maintain a database of application messages).
```

Comment:
Same here.

Reply:
ahh, from this comment are you saying that today Status Desktop already stores all messages received twice, once in status-go's local message database and it also keeps the received messages in a separate 'node' database?

We definitely do not want to delete messages from the owner node's status-go local message database

Reply:
Please see this comment: https://github.com/status-im/specs/pull/162/files#r767998941